### PR TITLE
Enhancing LinWei's Paging construct

### PR DESCRIPTION
**Pull request by honhaochen**
Apart from making my FinancialTrackerPage being able to switch between main and financial tracker page smoothly, I've also implemented ways to load FXML files easily. You guys can easily deal with SceneBuilder afterwards. With that being said, do look around and tell me if you think it's okay :P

Do check out my FinancialTrackerPage.java constructor

### CODE REVIEW

File: src/main/java/seedu/address/ui/Pages.java [hunk 1]
```diff
@@ -9,18 +9,19 @@
...
-    private Scene mainPageScene;
+    private static List<Page> pages;
+    private static Page mainPage;
 
     // todo: change this parameter to the specified scenes to minimise possible errors
     // keep PageScene constructor package private to prevent creation of this class elsewhere
-    Pages(Scene mainPageScene, Page... pages) {
-        this.mainPageScene = mainPageScene;
+    Pages(Page mainPage, Page ... pages) {
+        this.mainPage = mainPage;
```
Comment (bjhoohaha) [inline src/main/java/seedu/address/ui/Pages.java]: Can be refactored into if dont intend to use `Scene mainPageScene`. mainPage is also a page

```
Pages(Page ... pages) {
}
```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -68,13 +70,15 @@ public MainWindow(Stage primaryStage, Logic logic) {
         setAccelerators();
 
         helpWindow = new HelpWindow();
+        financialTrackerPage = new FinancialTrackerPage();
         calendarPage = new CalendarPage();
 
-        Scene primaryScene = primaryStage.getScene();
+        mainScene = primaryStage.getScene();
```
Comment (bjhoohaha) [inline src/main/java/seedu/address/ui/MainWindow.java]: Is the `mainScene` instance field required if we could simply use `primaryStage.getScene()` to get the scene?

Or are there other uses that I am not seeing?  ‎🤔

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -216,4 +221,14 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    @Override
+    public Scene getScene() {
+        return mainScene;
+    }
```
Comment (bjhoohaha) [inline src/main/java/seedu/address/ui/MainWindow.java]: Doesnt really need the mainScene instance field i feel...

```
@Override
public Scene getScene() {
     return primaryStage.getScene();
}
```

File: src/main/java/seedu/address/financialtracker/commands/DeleteFinCommand.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package seedu.address.financialtracker.commands;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * add financial expenses command for Financial Tracker.
+ */
```
Comment (bjhoohaha) [inline src/main/java/seedu/address/financialtracker/commands/DeleteFinCommand.java]: Update javadoc comments as well? 


```
/**
 * add financial expenses command for Financial Tracker.
 */
```

File: src/main/java/seedu/address/financialtracker/ui/FinancialTrackerPage.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    private StackPane resultDisplayPlaceholder;
+
+    @FXML
+    private StackPane statusbarPlaceholder;
+
+    public FinancialTrackerPage() {
+        this.financialTrackerParser = new FinancialTrackerParser();
+        fxmlLoader = new FXMLLoader(getFxmlFileUrl());
+        fxmlLoader.setController(this);
+        System.out.println(this);
```
[the remark this answers is not in the document]
Comment (honhaochen) [reply src/main/java/seedu/address/financialtracker/ui/FinancialTrackerPage.java]: ahh i didn't realise i left it there, was meant for my checkings

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 2 above)
Comment (honhaochen) [reply src/main/java/seedu/address/ui/MainWindow.java]: It's needed to preserve the mainScene. As primaryStage.getScene() returns whatever scene is currently on the stage, which could be other pages' scene

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (honhaochen) [reply src/main/java/seedu/address/ui/MainWindow.java]: that way you might not be getting mainScene if you are currently having others' scene (diary, itenary, etc.)

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 2 above)
Comment (bjhoohaha) [reply src/main/java/seedu/address/ui/MainWindow.java]: Ok. Now i understand why, thanks for the explanation!

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (bjhoohaha) [reply src/main/java/seedu/address/ui/MainWindow.java]: Ok! Thanks, great explanation.

**Pull request merged**